Actions/GetArtifactsForDeployment/GetArtifactsForDeployment.ps1

```
@@ -51,13 +52,31 @@ if ($artifactsVersion -eq "current" -or $artifactsVersion -eq "prerelease" -or $
         }
     }
 }
+elseif ($artifactsVersion -like "PR_*") {
+    $prId = $artifactsVersion -replace "PR_", ""
+    if (!($prId -as [int])) {
+        throw "Invalid PR id: $prId"
+    }
+    $prBranch = GetBranchFromPRId -repository $ENV:GITHUB_REPOSITORY -prId $prId -token $token
+    if (!($prBranch)) {
+        throw "Unable to locate branch for PR $prId"
+    }
+    $latestPRRun = FindLatestPRRun -repository $ENV:GITHUB_REPOSITORY -branch $prBranch -token $token
+    if ($latestPRRun -eq 0) {
+        throw "Latest PR build for PR $prId not found, not completed or not successful"
+    }
+
+    $artifactsToDownload | ForEach-Object {
+        GetArtifactsFromWorkflowRun -workflowRun $latestPRRun -token $token -api_url $ENV:GITHUB_API_URL -repository $ENV:GITHUB_REPOSITORY -mask $_ -projects $projects
+    }
+}
 else {
     $searchArtifacts = $true
 }
 
 if ($searchArtifacts) {
     $allArtifacts = @()
-    'Apps','Dependencies','PowerPlatformSolution' | ForEach-Object {
+    $artifactsToDownload | ForEach-Object {
```

What happens here if the artifacts from the PR build has expired?

We should throw an error stating that the build artifacts have expired, stating that people would need to re-run the pull-request build (and add a link to the PR Run). You could also add a note that the retention period for Short Lived Artifacts (such as PR builds) is controlled by the AL-Go setting [shortLivedArtifactsRetentionDays](https://aka.ms/algosettings#shortLivedArtifactsRetentionDays)

---

I added a new ref parameter to GetArtifactsFromWorkflowRun which adds any expired artifacts. We can potentially use that elsewhere too.